-------------------------------------------------------------------------------
Test set: org.apache.ratis.server.simulation.TestRaftWithSimulatedRpc
-------------------------------------------------------------------------------
Tests run: 7, Failures: 4, Errors: 0, Skipped: 0, Time elapsed: 44.038 s <<< FAILURE! - in org.apache.ratis.server.simulation.TestRaftWithSimulatedRpc
org.apache.ratis.server.simulation.TestRaftWithSimulatedRpc.testWithLoad  Time elapsed: 12.181 s  <<< FAILURE!
java.lang.AssertionError: Unexpected exit.
	at org.apache.ratis.util.ExitUtils.assertNotTerminated(ExitUtils.java:109)
	at org.apache.ratis.server.impl.MiniRaftCluster.shutdown(MiniRaftCluster.java:855)
	at org.apache.ratis.server.impl.MiniRaftCluster$Factory$Get.runWithNewCluster(MiniRaftCluster.java:146)
	at org.apache.ratis.server.impl.MiniRaftCluster$Factory$Get.runWithNewCluster(MiniRaftCluster.java:120)
	at org.apache.ratis.RaftBasicTests.testWithLoad(RaftBasicTests.java:343)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:299)
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:293)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.lang.Thread.run(Thread.java:750)
Caused by: org.apache.ratis.util.ExitUtils$ExitException: s3.serverHandler0 is terminating.
	at org.apache.ratis.util.ExitUtils.terminate(ExitUtils.java:141)
	at org.apache.ratis.util.ExitUtils.terminate(ExitUtils.java:151)
	at org.apache.ratis.server.simulation.RequestHandler$HandlerDaemon.run(RequestHandler.java:138)
Caused by: java.lang.IllegalStateException: entry.term: expected == 1 but computed == 2
	at org.apache.ratis.util.Preconditions.assertTrue(Preconditions.java:76)
	at org.apache.ratis.util.Preconditions.assertSame(Preconditions.java:81)
	at org.apache.ratis.statemachine.TransactionContext.wrap(TransactionContext.java:104)
	at org.apache.ratis.statemachine.impl.TransactionContextImpl.wrap(TransactionContextImpl.java:136)
	at org.apache.ratis.server.raftlog.segmented.SegmentedRaftLog.appendEntryImpl(SegmentedRaftLog.java:434)
	at org.apache.ratis.server.raftlog.RaftLogBase.lambda$appendEntry$11(RaftLogBase.java:351)
	at org.apache.ratis.server.raftlog.RaftLogSequentialOps$Runner.runSequentially(RaftLogSequentialOps.java:78)
	at org.apache.ratis.server.raftlog.RaftLogBase.appendEntry(RaftLogBase.java:351)
	at org.apache.ratis.server.raftlog.segmented.SegmentedRaftLog.appendImpl(SegmentedRaftLog.java:477)
	at org.apache.ratis.server.raftlog.RaftLogBase.lambda$append$12(RaftLogBase.java:358)
	at org.apache.ratis.server.raftlog.RaftLogSequentialOps$Runner.runSequentially(RaftLogSequentialOps.java:69)
	at org.apache.ratis.server.raftlog.RaftLogBase.append(RaftLogBase.java:358)
	at org.apache.ratis.server.impl.RaftServerImpl.appendEntriesAsync(RaftServerImpl.java:1653)
	at org.apache.ratis.server.impl.RaftServerImpl.appendEntriesAsync(RaftServerImpl.java:1517)
	at org.apache.ratis.server.impl.RaftServerImpl.appendEntries(RaftServerImpl.java:1500)
	at org.apache.ratis.server.impl.RaftServerProxy.appendEntries(RaftServerProxy.java:663)
	at org.apache.ratis.server.simulation.SimulatedServerRpc$1.handleRequest(SimulatedServerRpc.java:163)
	at org.apache.ratis.server.simulation.SimulatedServerRpc$1.handleRequest(SimulatedServerRpc.java:153)
	at org.apache.ratis.server.simulation.RequestHandler.handleRequest(RequestHandler.java:91)
	at org.apache.ratis.server.simulation.RequestHandler$HandlerDaemon.run(RequestHandler.java:125)

org.apache.ratis.server.simulation.TestRaftWithSimulatedRpc.testWithLoad  Time elapsed: 12.182 s  <<< FAILURE!
java.lang.AssertionError: Unexpected exit.
	at org.apache.ratis.util.ExitUtils.assertNotTerminated(ExitUtils.java:109)
	at org.apache.ratis.BaseTest.assertNoFailures(BaseTest.java:90)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.RunAfters.invokeMethod(RunAfters.java:46)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:33)
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:299)
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:293)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.lang.Thread.run(Thread.java:750)
Caused by: org.apache.ratis.util.ExitUtils$ExitException: s3.serverHandler0 is terminating.
	at org.apache.ratis.util.ExitUtils.terminate(ExitUtils.java:141)
	at org.apache.ratis.util.ExitUtils.terminate(ExitUtils.java:151)
	at org.apache.ratis.server.simulation.RequestHandler$HandlerDaemon.run(RequestHandler.java:138)
Caused by: java.lang.IllegalStateException: entry.term: expected == 1 but computed == 2
	at org.apache.ratis.util.Preconditions.assertTrue(Preconditions.java:76)
	at org.apache.ratis.util.Preconditions.assertSame(Preconditions.java:81)
	at org.apache.ratis.statemachine.TransactionContext.wrap(TransactionContext.java:104)
	at org.apache.ratis.statemachine.impl.TransactionContextImpl.wrap(TransactionContextImpl.java:136)
	at org.apache.ratis.server.raftlog.segmented.SegmentedRaftLog.appendEntryImpl(SegmentedRaftLog.java:434)
	at org.apache.ratis.server.raftlog.RaftLogBase.lambda$appendEntry$11(RaftLogBase.java:351)
	at org.apache.ratis.server.raftlog.RaftLogSequentialOps$Runner.runSequentially(RaftLogSequentialOps.java:78)
	at org.apache.ratis.server.raftlog.RaftLogBase.appendEntry(RaftLogBase.java:351)
	at org.apache.ratis.server.raftlog.segmented.SegmentedRaftLog.appendImpl(SegmentedRaftLog.java:477)
	at org.apache.ratis.server.raftlog.RaftLogBase.lambda$append$12(RaftLogBase.java:358)
	at org.apache.ratis.server.raftlog.RaftLogSequentialOps$Runner.runSequentially(RaftLogSequentialOps.java:69)
	at org.apache.ratis.server.raftlog.RaftLogBase.append(RaftLogBase.java:358)
	at org.apache.ratis.server.impl.RaftServerImpl.appendEntriesAsync(RaftServerImpl.java:1653)
	at org.apache.ratis.server.impl.RaftServerImpl.appendEntriesAsync(RaftServerImpl.java:1517)
	at org.apache.ratis.server.impl.RaftServerImpl.appendEntries(RaftServerImpl.java:1500)
	at org.apache.ratis.server.impl.RaftServerProxy.appendEntries(RaftServerProxy.java:663)
	at org.apache.ratis.server.simulation.SimulatedServerRpc$1.handleRequest(SimulatedServerRpc.java:163)
	at org.apache.ratis.server.simulation.SimulatedServerRpc$1.handleRequest(SimulatedServerRpc.java:153)
	at org.apache.ratis.server.simulation.RequestHandler.handleRequest(RequestHandler.java:91)
	at org.apache.ratis.server.simulation.RequestHandler$HandlerDaemon.run(RequestHandler.java:125)

org.apache.ratis.server.simulation.TestRaftWithSimulatedRpc.testOldLeaderNotCommit  Time elapsed: 5.758 s  <<< FAILURE!
java.lang.AssertionError: Unexpected exit.
	at org.apache.ratis.util.ExitUtils.assertNotTerminated(ExitUtils.java:109)
	at org.apache.ratis.server.impl.MiniRaftCluster.shutdown(MiniRaftCluster.java:855)
	at org.apache.ratis.server.impl.MiniRaftCluster$Factory$Get.runWithNewCluster(MiniRaftCluster.java:146)
	at org.apache.ratis.server.impl.MiniRaftCluster$Factory$Get.runWithNewCluster(MiniRaftCluster.java:120)
	at org.apache.ratis.RaftBasicTests.testOldLeaderNotCommit(RaftBasicTests.java:226)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:27)
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:299)
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:293)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.lang.Thread.run(Thread.java:750)
Caused by: org.apache.ratis.util.ExitUtils$ExitException: s3.serverHandler0 is terminating.
	at org.apache.ratis.util.ExitUtils.terminate(ExitUtils.java:141)
	at org.apache.ratis.util.ExitUtils.terminate(ExitUtils.java:151)
	at org.apache.ratis.server.simulation.RequestHandler$HandlerDaemon.run(RequestHandler.java:138)
Caused by: java.lang.IllegalStateException: entry.term: expected == 1 but computed == 2
	at org.apache.ratis.util.Preconditions.assertTrue(Preconditions.java:76)
	at org.apache.ratis.util.Preconditions.assertSame(Preconditions.java:81)
	at org.apache.ratis.statemachine.TransactionContext.wrap(TransactionContext.java:104)
	at org.apache.ratis.statemachine.impl.TransactionContextImpl.wrap(TransactionContextImpl.java:136)
	at org.apache.ratis.server.raftlog.segmented.SegmentedRaftLog.appendEntryImpl(SegmentedRaftLog.java:434)
	at org.apache.ratis.server.raftlog.RaftLogBase.lambda$appendEntry$11(RaftLogBase.java:351)
	at org.apache.ratis.server.raftlog.RaftLogSequentialOps$Runner.runSequentially(RaftLogSequentialOps.java:78)
	at org.apache.ratis.server.raftlog.RaftLogBase.appendEntry(RaftLogBase.java:351)
	at org.apache.ratis.server.raftlog.segmented.SegmentedRaftLog.appendImpl(SegmentedRaftLog.java:477)
	at org.apache.ratis.server.raftlog.RaftLogBase.lambda$append$12(RaftLogBase.java:358)
	at org.apache.ratis.server.raftlog.RaftLogSequentialOps$Runner.runSequentially(RaftLogSequentialOps.java:69)
	at org.apache.ratis.server.raftlog.RaftLogBase.append(RaftLogBase.java:358)
	at org.apache.ratis.server.impl.RaftServerImpl.appendEntriesAsync(RaftServerImpl.java:1653)
	at org.apache.ratis.server.impl.RaftServerImpl.appendEntriesAsync(RaftServerImpl.java:1517)
	at org.apache.ratis.server.impl.RaftServerImpl.appendEntries(RaftServerImpl.java:1500)
	at org.apache.ratis.server.impl.RaftServerProxy.appendEntries(RaftServerProxy.java:663)
	at org.apache.ratis.server.simulation.SimulatedServerRpc$1.handleRequest(SimulatedServerRpc.java:163)
	at org.apache.ratis.server.simulation.SimulatedServerRpc$1.handleRequest(SimulatedServerRpc.java:153)
	at org.apache.ratis.server.simulation.RequestHandler.handleRequest(RequestHandler.java:91)
	at org.apache.ratis.server.simulation.RequestHandler$HandlerDaemon.run(RequestHandler.java:125)

org.apache.ratis.server.simulation.TestRaftWithSimulatedRpc.testOldLeaderNotCommit  Time elapsed: 5.758 s  <<< FAILURE!
java.lang.AssertionError: Unexpected exit.
	at org.apache.ratis.util.ExitUtils.assertNotTerminated(ExitUtils.java:109)
	at org.apache.ratis.BaseTest.assertNoFailures(BaseTest.java:90)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.RunAfters.invokeMethod(RunAfters.java:46)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:33)
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:299)
	at org.junit.internal.runners.statements.FailOnTimeout$CallableStatement.call(FailOnTimeout.java:293)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.lang.Thread.run(Thread.java:750)
Caused by: org.apache.ratis.util.ExitUtils$ExitException: s3.serverHandler0 is terminating.
	at org.apache.ratis.util.ExitUtils.terminate(ExitUtils.java:141)
	at org.apache.ratis.util.ExitUtils.terminate(ExitUtils.java:151)
	at org.apache.ratis.server.simulation.RequestHandler$HandlerDaemon.run(RequestHandler.java:138)
Caused by: java.lang.IllegalStateException: entry.term: expected == 1 but computed == 2
	at org.apache.ratis.util.Preconditions.assertTrue(Preconditions.java:76)
	at org.apache.ratis.util.Preconditions.assertSame(Preconditions.java:81)
	at org.apache.ratis.statemachine.TransactionContext.wrap(TransactionContext.java:104)
	at org.apache.ratis.statemachine.impl.TransactionContextImpl.wrap(TransactionContextImpl.java:136)
	at org.apache.ratis.server.raftlog.segmented.SegmentedRaftLog.appendEntryImpl(SegmentedRaftLog.java:434)
	at org.apache.ratis.server.raftlog.RaftLogBase.lambda$appendEntry$11(RaftLogBase.java:351)
	at org.apache.ratis.server.raftlog.RaftLogSequentialOps$Runner.runSequentially(RaftLogSequentialOps.java:78)
	at org.apache.ratis.server.raftlog.RaftLogBase.appendEntry(RaftLogBase.java:351)
	at org.apache.ratis.server.raftlog.segmented.SegmentedRaftLog.appendImpl(SegmentedRaftLog.java:477)
	at org.apache.ratis.server.raftlog.RaftLogBase.lambda$append$12(RaftLogBase.java:358)
	at org.apache.ratis.server.raftlog.RaftLogSequentialOps$Runner.runSequentially(RaftLogSequentialOps.java:69)
	at org.apache.ratis.server.raftlog.RaftLogBase.append(RaftLogBase.java:358)
	at org.apache.ratis.server.impl.RaftServerImpl.appendEntriesAsync(RaftServerImpl.java:1653)
	at org.apache.ratis.server.impl.RaftServerImpl.appendEntriesAsync(RaftServerImpl.java:1517)
	at org.apache.ratis.server.impl.RaftServerImpl.appendEntries(RaftServerImpl.java:1500)
	at org.apache.ratis.server.impl.RaftServerProxy.appendEntries(RaftServerProxy.java:663)
	at org.apache.ratis.server.simulation.SimulatedServerRpc$1.handleRequest(SimulatedServerRpc.java:163)
	at org.apache.ratis.server.simulation.SimulatedServerRpc$1.handleRequest(SimulatedServerRpc.java:153)
	at org.apache.ratis.server.simulation.RequestHandler.handleRequest(RequestHandler.java:91)
	at org.apache.ratis.server.simulation.RequestHandler$HandlerDaemon.run(RequestHandler.java:125)

